db/engine/seed.js

```diff
@@ -0,0 +1,146 @@
+'use strict'
+
+const path = require('path')
+const fs = require('fs')
+const globalDotEnvFile = path.join(__dirname, '..', '..', '..', '.env')
+
+try {
+  if (fs.existsSync(globalDotEnvFile)) {
+    console.log('globalDotEnvFile found', globalDotEnvFile)
+    require('dotenv').config({ path: globalDotEnvFile })
+  } else {
+    console.log('No .env file found at project root')
+  }
+} catch (error) {
+  console.error(error)
+}
+const R = require('ramda')
+const csv = require('fast-csv')
+const logger = require('../../services/log.service').getLogger()
+
+const sqlService = require('../../services/data-access/sql.service')
+
+const seedsDirectory = path.join(__dirname, '/seeds')
+const seedFilenameFormat = ['version', 'table', 'name', 'format']
+const supportedFileFormats = ['tsv', 'sql', 'js']
+
+const loadSeeds = () => (new Promise((resolve, reject) => {
+  fs.readdir(seedsDirectory, (err, files) => {
+    if (err) {
+      return reject(err)
+    }
+    resolve(files)
+  })
+}))
+
+const processSeed = async (seed) => {
+  const { format, filename } = seed
+  const filepath = path.join(seedsDirectory, filename)
+
+  if (!supportedFileFormats.includes(format)) {
+    throw new Error(`Unsupported format: ${format} for seed ${filename}`)
+  }
+
+  let content; let sql; let params = []
+  if (format === 'tsv') {
+    // handle TSV
+    content = []
+    await new Promise((resolve, reject) => {
+      csv.parseFile(filepath, { delimiter: '\t', headers: true, trim: true })
+        .on('data', (data) => {
+          content.push(data)
+        })
+        .on('error', reject)
+        .on('end', resolve)
+    })
+    if (content.length === 0) {
+      logger.info(`Empty seed: ${filename}`)
+      return
+    }
+    const { table } = seed
+    const generatedSql = await sqlService.generateMultipleInsertStatements(table, content)
+    sql = generatedSql.sql
+    params = generatedSql.params
+  } else if (format === 'sql') {
+    // handle SQL
+    content = fs.readFileSync(filepath, 'utf8')
+    sql = content
+  } else if (format === 'js') {
+    // handle JS
+    content = require(filepath)
+    sql = await content.generateSql()
+  }
+
+  logger.info(filename)
+  try {
+    await sqlService.modify(sql, params)
+  } catch (error) {
+    /*
+      We can ignore certain error codes and
+      assume the seed is being re run
+    */
+    const ignoreErrorCodes = [
+      2601, // Cannot insert duplicate key
+      2627, // Violation of UNIQUE KEY constraint
+      547 // The INSERT statement conflicted with the FOREIGN KEY constraint
+    ]
+    if (!~ignoreErrorCodes.indexOf(error.number)) {
+      throw error
+    } else {
+      logger.warn(`ignoring error: ${error.message}`)
+      logger.warn('This seed is likely to have been previously applied to this database.')
+    }
```

Comment:
I think this is older code rather than something newly written.  Even so, this seems unclear.  Perhaps this would be clearer:

```
    if (ignoreErrorCodes.indexOf(error.number) >= 0) {
      logger.warn('This seed is likely to have been previously applied to this database.')
      logger.warn(`ignoring error: ${error.message}`)
    } else {
      throw error
    }
```

Am I missing anything?

Reply:
done.  nice to get rid of this